File: Examples/Perl/Document/SignPdfDoc.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposePdfCloud::PdfApi;
use AsposePdfCloud::ApiClient;
use AsposePdfCloud::Configuration;
use AsposePdfCloud::Object::Rectangle;
use AsposePdfCloud::Object::Signature;

my $configFile = '../config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposePdfCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposePdfCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposePdfCloud::Configuration::debug = 1;

$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Pdf API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $pdfApi = AsposePdfCloud::PdfApi->new();

# Set input file name
my $name =  'sample-input-2.pdf';
my $signatureFileName = 'pkc7-sample.pfx';
my @rect = AsposePdfCloud::Object::Rectangle->new('X' => 100, 'Y' => 100, 'Height' => 100, 'Width' => 200);
my @textReplaceListRequestBody = AsposePdfCloud::Object::Signature->new(
		'Authority' => 'Naeem Akram',
		'Location' => 'Gojra',
		'Contact' => 'naeem.akram@aspose.com',
		'Date' => '05/09/2016 2:00:00.000 AM',
		'FormFieldName' =>  'Signature1',
		'Password' => 'aspose',
		'SignaturePath' => $signatureFileName,
		'SignatureType' => 'PKCS7',
		'Visible' => 'True',
		'Rectangle' => @rect
);

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);
$response = $storageApi->PutCreate(Path => $signatureFileName, file => $data_path.$signatureFileName);

# Invoke Aspose.Pdf Cloud SDK API to sign PDF document                                             
$response = $pdfApi->PostSignDocument(name=>$name, body=>@textReplaceListRequestBody);

if($response->{'Status'} eq 'OK'){
	#download signed pdf from storage server
	my $output_file = $out_path. $name;
	$response = $storageApi->GetDownload(Path => $name);
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1

File: Examples/DotNET/CSharp/Document/SignPdfDoc.cs
using System;
using Com.Aspose.PDF.Api;
using Com.Aspose.PDF.Model;
using Com.Aspose.Storage.Api;

namespace Document
{
    class SignPdfDoc
    {
        public static void Run()
        {
            // ExStart:1
            PdfApi pdfApi = new PdfApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "sample-input-2.pdf";
            String signatureFileName = "pkc7-sample.pfx";            
            String storage = "";
            String folder = "";

            Signature body = new Signature();
            body.Authority = "Farooq Sheikh";
            body.Location = "Gojra";            
            body.Contact = "naeem.akram@aspose.com";
            body.Date = "06/20/2017 2:00:00.000 AM";
            body.FormFieldName = "Signature1";
            body.Password = "aspose";
            Rectangle rect = new Rectangle();
            rect.X = 100;
            rect.Y = 100;
            rect.Height = 100;
            rect.Width = 200;
            body.Rectangle = rect;
            body.SignaturePath = signatureFileName;
            body.SignatureType = "PKCS7";
            body.Visible = true;

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));
                storageApi.PutCreate(signatureFileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + signatureFileName));

                // Invoke Aspose.PDF Cloud SDK API to sign Pdf document                
                SaaSposeResponse apiResponse = pdfApi.PostSignDocument(fileName, storage, folder, body);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    // Download created pdf file
                    Com.Aspose.Storage.Model.ResponseMessage storageRes = storageApi.GetDownload(fileName, null, null);
                    // Save response stream to a file 
                    System.IO.File.WriteAllBytes(Common.GetDataDir() + "SignPdfDoc_out.pdf", storageRes.ResponseStream);    

                    Console.WriteLine("Sign PDF Documents, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}
